labview/labview_rabbitmq.c

```diff
@@ -5,251 +5,170 @@
-	if (status<0) {
-    return _OPENING_TCP_SOCKET;
-  	} 
+	if (status < 0)
+	{
+		return _OPENING_TCP_SOCKET;
+	}
+
 	/*Code explanation:
 	socket is set/stored in the connection state,
 	so we dont need to destroy this socket because
-	it will be destroyed along with connection state destroy function*/ 
-
-	char const *VHOST = "/"; 		// the virtual host to connect to on the broker. The default on most brokers is "/"
-	int const CHANNEL_MAX = 0; 		// the limit for number of channels for the connection. 0 means no limit.
-	int const FRAME_MAX = 131072; 	// the maximum size of an AMQP frame. 131072 is the default. 
-									// 4096 is the minimum size, 2^31-1 is the maximum, a good default is 131072 (128KB),
-	int const HEARTBEAT = 0; 		// the number of seconds between heartbeat frames to request of the broker. A value of 0 disables heartbeats.
-
-	return lv_report_amqp_error(amqp_login(conn, VHOST, CHANNEL_MAX, FRAME_MAX, HEARTBEAT, AMQP_SASL_METHOD_PLAIN, username, password), "Logging in", error_description);	
-}
-
-
-LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_basic_publish(int64_t conn_intptr, uint16_t  channel, char *exchange, char *routingkey, char *messagebody, LStrHandle error_description) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
-	amqp_basic_properties_t props;
-	props._flags = AMQP_BASIC_CONTENT_TYPE_FLAG | AMQP_BASIC_DELIVERY_MODE_FLAG;
-	props.content_type = amqp_cstring_bytes("text/plain");
-	props.delivery_mode = 2; /* persistent delivery mode */
-	return amqp_basic_publish(conn, channel, amqp_cstring_bytes(exchange),amqp_cstring_bytes(routingkey), 0, 0, &props, amqp_cstring_bytes(messagebody));
-	//this function returns amqp_status_enum thats different from amqp_rpc_reply_t
+	it will be destroyed along with connection state destroy function*/
+
+	char const *VHOST = "/";	// the virtual host to connect to on the broker.
+								// The default on most brokers is "/"
+	int const CHANNEL_MAX = 0;	// the limit for number of channels for the connection. 0 means no limit.
+	int const FRAME_MAX = 131072;	// the maximum size of an AMQP frame. 131072 is the default. 
+									// 4096 is the minimum size, 2^31-1 is the maximum,
+									// a good default is 131072 (128KB),
+	int const HEARTBEAT = 0;	// the number of seconds between heartbeat frames to request
+								// of the broker. A value of 0 disables heartbeats.
+
+	return lv_report_amqp_error(amqp_login(conn, VHOST, CHANNEL_MAX, FRAME_MAX, HEARTBEAT, AMQP_SASL_METHOD_PLAIN, username, password), "Logging in", errorDescription);
 }
 
-
 LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_create_queue(int64_t conn_intptr, uint16_t  channel, char *exchange, char *bindingkey, LStrHandle error_description) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+int lv_amqp_create_queue(int64_t conn_intptr, uint16_t channel, char* queue_name_in, LStrHandle queue_name_out, uint8_t passive, LStrHandle errorDescription)
```

Comment:
I see you added passive as parameters but it is not used in the function

Reply:
true, that was a part that was changed in the last day. I will simply update the repo. Sorry for not informing you.